src/main/java/org/spongepowered/common/data/SpongeDataManager.java

```diff
@@ -99,6 +114,10 @@
     private final Map<Class<? extends DataManipulator<?, ?>>, DataProcessorDelegate<?, ?>> dataProcessorDelegates =  new IdentityHashMap<>();
     private final Map<Class<? extends ImmutableDataManipulator<?, ?>>, DataProcessorDelegate<?, ?>> immutableDataProcessorDelegates =  new IdentityHashMap<>();
     private final Map<Class<? extends DataManipulator<?, ?>>, Class<? extends DataManipulator<?, ?>>> interfaceToImplDataManipulatorClasses = new IdentityHashMap<>();
+    private final Map<Key<? extends BaseValue<?>>, ContextValueProcessorDelegate<?, ?>> contextValueDelegates = new IdentityHashMap<>();
+    private final Map<Class<? extends DataManipulator<?, ?>>, ContextDataProcessorDelegate<?, ?>> contextDataProcessorDelegates =  new IdentityHashMap<>();
+    private final Map<Class<? extends ImmutableDataManipulator<?, ?>>, ContextDataProcessorDelegate<?, ?>> immutableContextDataProcessorDelegates =  new IdentityHashMap<>();
+    private final Map<Class<? extends DataManipulator<?, ?>>, Class<? extends DataManipulator<?, ?>>> contextInterfaceToImplDataManipulatorClasses = new IdentityHashMap<>();
```

Comment:
I don't think this is needed since you don't have to go between any sort of `DataManipulator` to `DataManipulator` class types for context.